modules/schematics/src/effect/files/__name@dasherize@if-flat__/__name@dasherize__.effects.ts

```diff
@@ -1,10 +1,33 @@
 import { Injectable } from '@angular/core';
 import { Actions, Effect<% if (feature) { %>, ofType<% } %> } from '@ngrx/effects';
-<% if (feature) { %>import { <%= classify(name) %>ActionTypes } from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
+<% if (feature && !api) { %>import { <%= classify(name) %>ActionTypes } from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
+<% if (feature && api) { %>import { catchError, map, switchMap } from 'rxjs/operators';<% } %>
+<% if (feature && api) { %>import { Load<%= classify(name) %>sFailure, Load<%= classify(name) %>sSuccess, <%= classify(name) %>ActionTypes } from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
 
 @Injectable()
 export class <%= classify(name) %>Effects {
-<% if (feature) { %>
+<% if (feature && api) { %>
+
+  private _testObservable = of([
+    { myDataProperty: 1000, myOtherDataProperty: 'some-text' },
+    { myDataProperty: 1100, myOtherDataProperty: 'more-text' }]
+  )
+    .pipe(
+      delay(1000), // simulate latency of 1 second
+      // (_ => { throw new Error('oops'); }) // uncomment to test error handling
+    );
+
+  @Effect()
+  load<%= classify(name) %>s$ = this.actions$.pipe(
+    ofType(<%= classify(name) %>ActionTypes.Load<%= classify(name) %>),
```

Comment:
With NgRx 7 we don't need to type the `ofType` operator, should we use the new syntax in the schematics?

It would be:

```
ofType(LoadFoos),

// and the actions defined as:
constructor(private actions$: Actions<FooActions>) {}
```

Reply:
@timdeschryver 
I will update the constructor, but the Effect should already output the following:
```typescript
 @Effect()
  loadFoos$ = this.actions$.pipe(
    ofType(FooActionTypes.LoadFoos),
    switchMap(() =>
      EMPTY.pipe(
        map(data => new LoadFoosSuccess({ data })),
        catchError(error => of(new LoadFoosFailure({ error })))
      )
    )
  );
```
Is that not correct?

Reply:
This should be resolved now.